### PR TITLE
User persistence

### PR DESCRIPTION
**Pull request by jjojan**
1. added user persistence to see stores in offline mode - for 1-3 : https://imgur.com/a/rtkV1SI
2. added user persistence to search stores in offline mode
3. added user persistence to see cart w/o checkout in offline mode 
4. added user persistence to see store pages w/o adding to cart function in offline mode
5. Added checkout animation - https://imgur.com/a/1BJBnQe

### CODE REVIEW

File: app/src/main/java/com/example/harvestfresh/CartRoom.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        for (CartRoom cartRoom: cartsFromRooms) {
+            Cart cart = new Cart();
+            ProductListing listing = new ProductListing();
+            listing.setProductPrice(cartRoom.productCost);
+            listing.setProductName(cartRoom.productName);
+            cart.setProduct(listing.getProductName());
+            cart.setPrice(listing);
+            carts.add(cart);
+        }
+    return carts;
```
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/CartRoom.java:46]: nit: off indentation.

File: app/src/main/java/com/example/harvestfresh/HarvestFreshDatabaseApplication.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import androidx.sqlite.db.SupportSQLiteOpenHelper;
+
+public class HarvestFreshDatabaseApplication extends Application {
+    HarvestFreshDatabase myDatabase;
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
+        myDatabase = Room.databaseBuilder(this, HarvestFreshDatabase.class, HarvestFreshDatabase.NAME).fallbackToDestructiveMigration().build();
```
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/HarvestFreshDatabaseApplication.java:18]: multi-line

File: app/src/main/java/com/example/harvestfresh/fragments/CartFragment.java [hunk 3]
```diff
@@ -108,9 +154,16 @@ public void onChanged() {
         btnCheckout.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                cartCheckout();
+                if (isNetworkConnected()){
+                    cartCheckout();
+                }
+                else {
```
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/fragments/CartFragment.java:160]: Indentation off.

File: app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java [hunk 4]
```diff
@@ -149,6 +160,41 @@ public void onError(@NonNull Status status) {
...
+        Fragment fragment = new HomeFragment();
+        fragmentManager.beginTransaction().replace(R.id.fLayoutContainer, fragment).commit();
+
+    }
+
+    private void animationControl(LottieAnimationView rvLoading) {
+        rvLoading.playAnimation();
+        rvLoading.addAnimatorListener(new Animator.AnimatorListener() {
+            @Override
+            public void onAnimationStart(Animator animation) {
```
Comment (rahul-pl) [inline app/src/main/java/com/example/harvestfresh/fragments/EventsFragment.java:179]: This and the rest of the calls should be replaced as -
```
public void onAnimationStart(Animator animation) {}
```

**Review by rahul-pl — APPROVED** [review]
1. I see that the animation is unrelated to the "User Persistence" story, could it have been in a separate PR ?
2. Also seeing quite a few indentation issues, any clue if they could have been caught by the linter ?

**Comment by jjojan** [comment]
> 1. I see that the animation is unrelated to the "User Persistence" story, could it have been in a separate PR ?
> 2. Also seeing quite a few indentation issues, any clue if they could have been caught by the linter ?

For the animation I intended to include in in a separate PR but I believe it got included with the user persistence since it was unmerged commit, sorry about that.

For the linter, I'm not sure - I think it should have though, I'm gonna run the files with the linter again just to be sure.

**Pull request merged**